packages/schematics/angular/migrations/update-8/update-lazy-module-paths.ts

```diff
@@ -5,26 +5,76 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
-import {
-  Rule,
-  SchematicContext,
-  Tree,
-} from '@angular-devkit/schematics';
-import {
-  TslintFixTask,
-} from '@angular-devkit/schematics/tasks';
-import * as path from 'path';
-
-export const updateLazyModulePaths = (): Rule => {
-  return (_: Tree, context: SchematicContext) => {
-    context.addTask(new TslintFixTask({
-      rulesDirectory: path.join(__dirname, 'rules'),
-      rules: {
-        'no-lazy-module-paths': [true],
-      },
-    }, {
-      includes: '**/*.ts',
-      silent: false,
-    }));
+import { DirEntry, Rule, UpdateRecorder } from '@angular-devkit/schematics';
+import * as ts from '../../third_party/github.com/Microsoft/TypeScript/lib/typescript';
+
+function* visit(directory: DirEntry): IterableIterator<ts.SourceFile> {
```

Comment:
Really nice use of a generator 👍 

This should reduce TS source files in memory to a constant 1.